File: ios/Classes/constants/AppConstants.swift
class AppConstants{
    
    public static let EMPTY = ""
    // method
    public static let METHOD_CHANNEL = "flutter_parental_control_method"
    public static let GET_DEVICE_INFO = "getDeviceInfoMethod"
    public static let GET_LOCATION = "getLocationMethod"
    public static let CHECK_PERMISSION = "permissionMethod"
    public static let SCHEDULE_MONITOR = "scheduleMonitorMethod"
    public static let LIMITED_APP = "limitAppMethod"
    public static let SETTING_MONITOR = "settingMonitorMethod"
    
    // key data
    // device info
    public static let APPLE = "Apple"
    public static let UNKNOWN = "Unknown"
    public static let SYSTEM_NAME = "systemName"
    public static let DEVICE_NAME = "deviceName"
    public static let DEVICE_MANUFACTURER = "deviceManufacturer"
    public static let SYSTEM_VERSION = "systemVersion"
    public static let DEVICE_API_LEVEL = "deviceApiLevel"
    public static let BATTERY_LEVEL = "batteryLevel"
    public static let SCREEN_BRIGHTNESS = "screenBrightness"
    public static let VOLUME = "volume"
    public static let MODEL_NAME = "modelName"
    public static let LOCALIZED_MODEL = "localizedModel"
    public static let DEVICE_ID = "deviceId"
    // key schedule monitor
    public static let DAILY = "daily"
    public static let WEEKLY = "weekly"
    public static let IS_MONITORING = "isMonitoring"
    public static let START_HOUR = "startHour"
    public static let START_MINUTE = "startMinute"
    public static let END_HOUR = "endHour"
    public static let END_MINUTE = "endMinute"
    // key monitor setting
    public static let REQUIRE_AUTO_DATE = "requireAutomaticDateAndTime"
    public static let LOCK_ACCOUNTS = "lockAccounts"
    public static let LOCK_PASSCODE = "lockPasscode"
    public static let DENY_SIRI = "denySiri"
    public static let LOCK_APP_CELLULAR_DATA = "lockAppCellularData"
    public static let LOCK_E_SIM = "lockESIM"
    public static let DENY_IN_APP_PURCHASES = "denyInAppPurchases"
    public static let MAXIMUM_RATING = "maximumRating"
    public static let REQUIRE_PASSWORD_FOR_PURCHASES = "requirePasswordForPurchases"
    public static let DENY_EXPLICIT_CONTENT = "denyExplicitContent"
    public static let DENY_MUSIC_SERVICE = "denyMusicService"
    public static let DENY_BOOKSTORE_EROTICA = "denyBookstoreErotica"
    public static let MAXIMUM_MOVIE_RATING = "maximumMovieRating"
    public static let MAXIMUM_TV_SHOW_RATING = "maximumTVShowRating"
    public static let DENY_MULTIPLAYER_GAMING = "denyMultiplayerGaming"
    public static let DENY_ADDING_FRIENDS = "denyAddingFriends"
    
    // error code
    public static let UNAVAILABLE = "UNAVAILABLE"
    public static let AUTH_NOT_DETERMINED = "AUTH_NOT_DETERMINED"
    public static let AUTH_DENIED = "AUTH_DENIED"
    public static let AUTH_APPROVED = "AUTH_APPROVED"
    public static let UNKNOWN_AUTH_STATUS = "UNKNOWN_AUTH_STATUS"
    public static let NO_ICLOUD_ACCOUNT = "NO_ICLOUD_ACCOUNT"
    public static let INVALID_ACCOUNT_TYPE = "INVALID_ACCOUNT_TYPE"
    public static let AUTHORIZATION_CANCELED = "AUTHORIZATION_CANCELED"
    public static let UNKNOWN_ERROR = "UNKNOWN_ERROR"
    public static let SCHEDULE_MONITOR_ERROR = "SCHEDULE_MONITOR_ERROR"
    
    // error
    public static let CONTROLLER_ERROR = "FlutterViewController not available"
    public static let SCHEDULE_ERROR =  "Lỗi khi lên lịch giám sát"
}

File: lib/src/core/app_constants.dart
class AppConstants {
  static const String empty = '';

  /// Method
  static const String methodChannel = 'flutter_parental_control_method';
  static const String eventChannel = 'flutter_parental_control_event';
  static const String deviceMethod = 'getDeviceInfoMethod';
  static const String appDetailMethod = 'getAppDetailMethod';
  static const String deviceUsageMethod = 'getDeviceUsageMethod';
  static const String locationMethod = 'getLocationMethod';
  static const String appUsageMethod = 'getAppUsageInfoMethod';
  static const String getToDayUsage = 'getTodayUsageMethod';
  static const String blockAppMethod = 'blockAppMethod';
  static const String blockWebMethod = 'blockWebsiteMethod';
  static const String startServiceMethod = 'startServiceMethod';
  static const String permissionMethod = 'permissionMethod';
  static const String scheduleMethod = 'scheduleMonitorMethod';
  static const String limitAppMethod = 'limitAppMethod';
  static const String settingMonitorMethod = 'settingMonitorMethod';
  static const String getWebHistory = 'getWebHistoryMethod';
  static const String overlayMethod = 'overlayMethod';
  static const String askParentMethod = 'askParentMethod';

  /// key data channel
  static const String blockApps = 'blockApps';
  static const String blockWeb = 'blockWebsites';
  static const String typePermission = 'typePermission';

  /// key map
  static const String systemName = 'systemName';
  static const String deviceName = 'deviceName';
  static const String deviceManufacturer = 'deviceManufacturer';
  static const String deviceVersion = 'deviceVersion';
  static const String deviceApiLevel = 'deviceApiLevel';
  static const String deviceBoard = 'deviceBoard';
  static const String deviceHardware = 'deviceHardware';
  static const String deviceDisplay = 'deviceDisplay';
  static const String batteryLevel = 'batteryLevel';
  static const String screenBrightness = 'screenBrightness';
  static const String volume = 'volume';
  static const String deviceId = 'deviceId';

  static const String packageName = 'packageName';
  static const String appName = 'appName';
  static const String appIcon = 'appIcon';
  static const String versionName = 'versionName';
  static const String versionCode = 'versionCode';
  static const String timeInstall = 'timeInstall';
  static const String timeUpdate = 'timeUpdate';
  static const String isInstalled = 'isInstalled';
  static const String usageTime = 'usageTime';
  static const String timeLimit = 'timeLimit';
  static const String timeUsed = 'timeUsed';

  static const String name = 'name';
  static const String street = 'street';
  static const String locality = 'locality';
  static const String subLocality = 'subLocality';
  static const String subAdminArea = 'subAdminArea';
  static const String adminArea = 'adminArea';
  static const String country = 'country';

  static const String searchQuery = 'searchQuery';
  static const String visitedTime = 'visitedTime';

  static const String id = 'id';
  static const String overlayView = 'overlayView';
  static const String backBtn = 'backButton';
  static const String askParentBtn = 'askParentBtn';
  static const String day = 'day';

  /// cài đặt thời gian hạn chế [ios]
  static const String isMonitoring = 'isMonitoring';
  static const String startHour = 'startHour';
  static const String startMinute = 'startMinute';
  static const String endHour = 'endHour';
  static const String endMinute = 'endMinute';

  /// các cài đặt hạn chế [ios]
  static const String requireAutoTime = 'requireAutomaticDateAndTime';
  static const String lockAccounts = 'lockAccounts';
  static const String lockPasscode = 'lockPasscode';
  static const String denySiri = 'denySiri';
  static const String lockAppCellularData = 'lockAppCellularData';
  static const String lockESIM = 'lockESIM';
  static const String denyInAppPurchases = 'denyInAppPurchases';
  static const String maximumRating = 'maximumRating';
  static const String requirePasswordForPurchases =
      'requirePasswordForPurchases';
  static const String denyExplicitContent = 'denyExplicitContent';
  static const String denyMusicService = 'denyMusicService';
  static const String denyBookstoreErotica = 'denyBookstoreErotica';
  static const String maximumMovieRating = 'maximumMovieRating';
  static const String maximumTVShowRating = 'maximumTVShowRating';
  static const String denyMultiplayerGaming = 'denyMultiplayerGaming';
  static const String denyAddingFriends = 'denyAddingFriends';

  /// error
  static const String iosPlatformError = 'iOS platform specific function';
  static const String androidPlatformError =
      'Android platform specific function';
  static const String addressError = 'Unable to get location information';
  static const String locationError = 'Location access required';
  static const String safeZoneError = 'Safe Zone must have 3 points or more';
}
